LLama/Transform/StreamingTokenDecoder.cs

```diff
@@ -6,14 +6,14 @@
 using LLama.Extensions;
 using LLama.Native;
 
-namespace LLama
+namespace LLama.Transform
 {
     /// <summary>
     /// Decodes a stream of tokens into a stream of characters
     /// </summary>
     public sealed class StreamingTokenDecoder
     {
-        private readonly SafeLlamaModelHandle _weights;
+        private readonly SafeLlamaModelHandle? _weights;
```

Comment:
I don't think any of the changes in this class are necessary if we go with the suggested change to the `DefaultTokenizer`